x-pack/plugins/enterprise_search/public/applications/analytics/components/analytics_overview/analytics_collections_logic.ts

```diff
@@ -16,39 +16,69 @@ import {
 } from '../../api/index/fetch_analytics_collections_api_logic';
 
 export interface AnalyticsCollectionsActions {
-  fetchAnalyticsCollections(): void;
+  fetchAnalyticsCollections(query?: string): { query: string };
   makeRequest: Actions<{}, FetchAnalyticsCollectionsApiLogicResponse>['makeRequest'];
 }
 export interface AnalyticsCollectionsValues {
   analyticsCollections: AnalyticsCollection[];
   data: typeof FetchAnalyticsCollectionsAPILogic.values.data;
   hasNoAnalyticsCollections: boolean;
-  isLoading: boolean;
+  isFetching: boolean;
+  isSearchRequest: boolean;
+  isSearching: boolean;
+  searchQuery: string;
   status: Status;
 }
 
 export const AnalyticsCollectionsLogic = kea<
   MakeLogicType<AnalyticsCollectionsValues, AnalyticsCollectionsActions>
 >({
   actions: {
-    fetchAnalyticsCollections: () => {},
+    fetchAnalyticsCollections: (query) => ({
+      query,
+    }),
   },
   connect: {
     actions: [FetchAnalyticsCollectionsAPILogic, ['makeRequest']],
     values: [FetchAnalyticsCollectionsAPILogic, ['data', 'status']],
   },
   listeners: ({ actions }) => ({
-    fetchAnalyticsCollections: () => {
-      actions.makeRequest({});
+    fetchAnalyticsCollections: async ({ query }, breakpoint) => {
+      if (query) {
+        await breakpoint(200);
+      }
+      actions.makeRequest({ query });
     },
   }),
   path: ['enterprise_search', 'analytics', 'collections'],
+  reducers: {
+    isSearchRequest: [
```

Comment:
do we still need isSearchRequest, now that we have `searchQuery`?

Reply:
We can avoid, but in that case we need to put some logic in query. Like if it's null(because we can't have undefined) that means we just fetched items, if it's empty that means we clear search. I would  even split in two functions, like search and fetch, to make it more obvious. Because now when we use this functions we need to keep in mind logic besides it